django/cantusdb_project/main_app/tests/test_views.py

```diff
@@ -152,3 +154,134 @@ def test_view_context_data(self):
             response = self.client.get(reverse("indexer-detail", args=[indexer.id]))
             self.assertTrue("indexer" in response.context)
             self.assertEquals(indexer, response.context["indexer"])
+
+
+class FeastListViewTest(TestCase):
+    PAGE_SIZE = FeastListView.paginate_by
+    fixtures = ["feast_fixtures.json"]
+
+    def setUp(self):
+        self.number_of_feasts = Feast.objects.all().count()
+        return super().setUp()
+
+    def test_view_url_path(self):
+        response = self.client.get("/feasts/")
+        self.assertEqual(response.status_code, 200)
+
+    def test_view_url_reverse_name(self):
+        response = self.client.get(reverse("feast-list"))
+        self.assertEqual(response.status_code, 200)
+
+    def test_view_correct_templates(self):
+        response = self.client.get(reverse("feast-list"))
+        self.assertEqual(response.status_code, 200)
+        self.assertTemplateUsed(response, "base.html")
+        self.assertTemplateUsed(response, "feast_list.html")
+
+    # TODO: maybe make a more general method to test pagination that I can apply
+    # to all list views?
+    def test_pagination(self):
+        # To get total number of pages do a ceiling integer division
+        q, r = divmod(self.number_of_feasts, self.PAGE_SIZE)
+        pages = q + bool(r)
+
+        # Test all pages
+        for page_num in range(1, pages + 1):
+            response = self.client.get(reverse("feast-list"), {"page": page_num})
+            self.assertEqual(response.status_code, 200)
+            self.assertTrue("is_paginated" in response.context)
+            self.assertTrue(response.context["is_paginated"] == True)
+            if page_num == pages and (self.number_of_feasts % self.PAGE_SIZE != 0):
+                self.assertTrue(
+                    len(response.context["feasts"])
+                    == (self.number_of_feasts % self.PAGE_SIZE)
+                )
```

Comment:
`self.assertEqual` might be more descriptive code.
```suggestion
                self.assertEqual(
                    len(response.context["feasts"]),
                    self.number_of_feasts % self.PAGE_SIZE
                )
```